### ISSUE TITLE
Maximum dimensions and weight of a carrier

### ISSUE THREAD

**Issue by paulnoelcholot** [PrestaShop/PrestaShop#36830]
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md).
- [X] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add attachments

If my carrier has a max order size of 100x100x100 and a max order weight of 50kg
If I take an item of 50x50x50 and a weight of 1kg: The carrier is available (and it's OK)
On the other hand...
If I choose to put 10 items like this in my order: The carrier is available (and it's NOK because the dimensions of the order exceed the max dimensions of the carrier)
If I choose to put 60 items like this in my order: The carrier is no longer available (and it's OK because the weight of the order exceeds the max weight of the carrier)

### Expected behavior

If the dimensions are for a product and not an order, this should be noted.
If the dimensions are for an order, the carrier should no longer be displayed.

### Steps to reproduce

1. On BO create a standart product
Name : test product carrier
Online
Stock : 50
Width : 50
Height : 50
Depth : 50
Weight : 1
Price (tax ecl) : 10€

2. Create a carrier : 
Free shipping = NO
Tax = No tax 
Handling costs = NO  
Shipping costs : Based on the order's total weight
Out-of-range behavior  = Disable carrier
Maximum package width (cm) = 100
Maximum package height (cm) = 100
Maximum package depth (cm) = 100
Maximum package weight (kg) = 50

3. On FO > Add the product to the cart
4. Modify the quantity : 20
5. Checkout
6. Carrier diplayed

### PrestaShop version(s) where the bug happened

8.1.x

### PHP version(s) where the bug happened

8.1

**Comment by kpodemski** [comment]
It's impossible to correctly restrict carriers on dimensions without using advanced techniques of calculation. It depends on how you stack your items, what fillers do you use, and how you secure the goods...

That being said, in my opinion it's not a bug :)

A more realistic example of your calculations:

### Carrier

Length: max 100
Width: max 100
Height: max 100

### Your cart
-----

**Product 1**

100 | 1 | 1

**Product 2**

100 | 1 | 1

**Product 3**

100 | 1 | 1

Now, let's discuss whether this cart fits the package or not.

**Comment by Hlavtox** [comment]
I agree with Krystian. I get your point @paulnoelcholot, but these restrictions are very tricky. And you can always send it in two packages. 👍

**Comment by paulnoelcholot** [comment]
FYI @PrestaShop/product-council
